### PR TITLE
Basic version for initial user information

### PR DESCRIPTION
**Pull request by umbobabo**
I need this initial prototype for the Omniture plugin but obviously should be a reusable class to get the information regarding the user for the cookie.

To test it add this row to your host file:

127.0.0.1 blog.economist.com

Open 2 different browsers and in one log-in into the economist website and do not log-in in the second one.
After running the example open the bowser at blog.economist.com:<PortUsedByTheLocalServer> and see the difference.

I keep the tech debt to add test.

### CODE REVIEW

File: index.es6 [hunk 1]
```diff
@@ -0,0 +1,39 @@
+import Cookie from '@economist/cookie';
+
+const cookie = new Cookie();
+export default class User {
```
Comment (keithamus) [inline index.es6]: No point in this being a class as far as I can see.

File: package.json [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+    "mocha": "^2.2.5",
+    "npm-assets": "^0.1.0",
+    "npm-watch": "0.0.1",
+    "parallelshell": "^2.0.0",
+    "pre-commit": "^1.0.10",
+    "react": "^0.14.2",
+    "watchify": "^3.4.0"
+  },
+  "peerDependencies": {
+    "react": "^0.14.0"
```
Comment (keithamus) [inline package.json]: I'm not sure this is a peer dependency of this project.

File: index.es6 [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+  splitValues(list, separator = ';') {
+    const splitObj = {};
+    list.split(separator)
+    .map((subStr) => {
+      let key = '';
+      let val = '';
+      [ key, val ] = subStr.trim().split('=').map(decodeURIComponent);
```
Comment (fabiosantoscode) [inline index.es6]: Beautiful. You can also do:

`const [ key = '', val = '' ] = ...`

And declare the variables instantly, if you like.

File: package.json (hunk 2 above)
Comment (fabiosantoscode) [inline package.json]: +1. If this is not going to use React other than for the example, this is merely a devDependency.

File: index.es6 (hunk 1 above)
Comment (fabiosantoscode) [inline index.es6]: +1. This should be an object-creating `function makeUser({ cookie = newCookie })`.

File: test/index.es6 [hunk 4]
```diff
@@ -0,0 +1,3 @@
+describe('', () => {
+
+});
```
Comment (fabiosantoscode) [inline test/index.es6]: There should be some tests here. It shouldn't be too hard to test that this works, no?

File: index.es6 (hunk 1 above)
Comment (keithamus) [inline index.es6]: I just don't see the point in any instantiation whatsoever. These are just singleton functions (and intentionally so), so no need to have class.

File: index.es6 (hunk 3 above)
Comment (keithamus) [inline index.es6]: In fact this whole function could be trimmed a bit by using Array#reduce:

```js
splitValues(list, separator = ';') {
  return list.split(separator)
    .reduce((obj, subStr) => {
      const [ key = '', val = '' ] = subStr.trim().split('=').map(decodeURIComponent);
      obj[key] = val;
      return obj
    }, {});
}
```

File: index.es6 [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      const allEntitlements = this.splitValues(cookie.getCookie('mmcore.uat'), ';');
+      requested = allEntitlements;
+      if (entitlement) {
+        requested = allEntitlements[entitlement];
+      }
+    } else {
+      requested = false;
+    }
+    return requested;
+  }
```
Comment (keithamus) [inline index.es6]: Do you need `requested` at all? It looks like not really.

```js
getEntitlements(entitlement) {
  if (cookie.getCookie('mmcore.uat')) {
    const allEntitlements = this.splitValues(cookie.getCookie('mmcore.uat'), ';');
    return (entitlement ? allEntitlements[entitlement] || {} : allEntitlements);
  }
  return false;
}
```

File: index.es6 (hunk 1 above)
Comment (fabiosantoscode) [inline index.es6]: Pass the cookie to an object or class instance or keep it as the default argument to many functions, either way is okay.

File: index.es6 (hunk 5 above)
Comment (umbobabo) [inline index.es6]: Your solution is sure more elegant and I will go for that, not sure will pass the lint validation, that is the reason I write as is.

**Comment by umbobabo** [comment]
@keithamus I wrote tests with @fabiosantoscode and reviewed all the note, pls review and merge if you are happy with this implementation.

**Comment by keithamus** [comment]
Trying to run the tests, but they don't run. How did you get them running in Karma?

**Comment by fabiosantoscode** [comment]
We ran them in the docs, haven't tested the tests (heh) in karma.

**Comment by fabiosantoscode** [comment]
@keithamus I just tried it



I tried some stuff for a while, and it turns out you need to run `npm run docs` before. That's being run already in `build.sh` but not on `pretest`, so I think a change to devpack needs to happen, lest more people trip and think tests don't run.

**Comment by umbobabo** [comment]
@keithamus @fabiosantoscode Can we merge?

**Comment by fabiosantoscode** [comment]
:rainbow:

**Pull request merged**